lib/controller-inputs/FrameController.h

```diff
@@ -0,0 +1,141 @@
+#pragma once
+
+#include "AnalogInputGroup.h"
+#include "DigitalInputGroup.h"
+#include "mbed.h"
+
+/*
+  When sending controller board data to the PC, we will use a custom protocol with two different
+  frames for digital and analog inputs respectively.
+  Digital inputs frame (one frame per digital inputs group):
+    SOF(8 bits, 0b10101010, i.e. 0xAA),
+    Board Selection: 2 bits, because 3 subboards,
+    Control element type selection: 2 bits, because 2 type of controls, pushbutton and switch,
+                                    the other bit is for future expansion,
+    Placeholder: 4 bits,
```

Comment:
Why don't you save a packet and use 4 bits for SOF/identification, 2 board select bits, 2 control element bits?

Reply:
Good point, I was being dumb lol